File: Code/Python/problem006/euler006.py
#!/usr/bin/env python
import numpy as np


def sum_of_squares(max):
    """square the numbers from 1 to max, inclusive, and sum them"""
    return np.square(np.arange(1, max + 1)).sum()


def square_of_sum(max):
    """square the sum of the numbers from 1 to max, inclusive"""
    return np.square(np.arange(1, max + 1).sum())


def euler(max):
    return square_of_sum(max) - sum_of_squares(max)


if __name__ == '__main__':
    print "The answer is %d" % euler(100)

File: Code/R/problem006/euler006.R
#!/usr/bin/env Rscript

sum.of.squares <- function(n) {
  if (n <= 0) {
    0
  }
  else {
    vv <- 1:n
    sum(sapply(vv, function(x){x^2}))
  }
}

square.of.sums <- function(n) {
  x <- sum(1:n)
  x^2
}

diff.of.squares <- function(n) {
  sqos <- square.of.sums(n)
  sosq <- sum.of.squares(n)
  sqos - sosq
}


